### ISSUE TITLE
Option to disable vehicle changing lane near the edge of the map

### ISSUE THREAD

**Issue by pcfantasy** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#83]
![screenshot 24](https://user-images.githubusercontent.com/22406133/52788824-1281b580-309d-11e9-8fd5-877d4531b284.png)

As the picture above, if there is large amout of traffic near the edge of the map, vehicles will like to change lane in the first node, and cause congestion.

There is a workaround: use laneconnector to make the first node(and the second.. the third...) to stay on line (disable lane changing), traffic congestion will be much better.

So is there anyway to disable lane changing  near the edge of the map, I am not familiar with routingmanager logic.

Tips: How to define the edge of the map, we can get segment position, and if abs(position.x > 8000) | abs(position.z > 8000), this segment  is near the edge of the map

**Comment by aubergine10** [comment]
tl;dr: just fine the outside connection spawn/despawn code and refactor it in to something awesome

If we're going to do something with the way vehicles spawn/despawn at the edge of the map, why not do a proper job of it?

So instead of pedestrians walking on to road, then doing a pocket car, an actual car would drive on to the road. And instead of a fake train station at the edge of the map, trains with the relevant cims would just drive on to and off of the map.

As a user of 81 tiles mod, I regularly build in the tiles near the edge of the map, and have disabled edge fog. Seeing the weird antics vehicles perform near outside connections is a constant annoyance.

**Comment by aubergine10** [comment]
A Steam user was mentioning about train traffic jams on edge of map. Eradicating those crazy fake train stations would help solve that issue too!

All the maps I've seen use two-lane two-way train tracks for the outside connections so in theory there's never any need to have the station or the station tracks crossover thing. One lane of the train track is for incoming, the other for outgoing.

Problems would arise if user added some single-lane or one-way train track to outside connection though. Same would be true of roads btw (eg. if using BPs "Any Road Outside Connection" mod).

**Comment by aubergine10** [comment]
Random ponder:

* Have a lane index pointer for each direction (entry/exit - as applicable) of an outside connection.
* Spawnings (entering map):
    * Cims wait at side of road (like fake bus stop)
    * When vehicle spawns, it spawns in to lane that the index pointer is currently at
        * index pointer += 1; if > lanes, index pointer = 0 (wraps back to first lane)
        * This way vehicle spawning is distributed across all the lanes
    * Rather than cims walk to vehicle, just teleport them in to it
* Despawnings (exiting map)
    * The vehicle is assigned exit lane that the index pointer is currently at
        * Increment pointer, wrap to 0 if necessary
    * This "roughly" distributes leaving traffic across all lanes (won't work very well as they will all take different time to get there)
* Have a flag that disables lane changes (by pathfinder or DLS) within certain distance of outside connection

**Comment by krzychu124** [comment]
comment to ponders :)
- despawn - not a big issue if you turn on DLS
- spawn 
    - fake busstop may not work since there is no pedestrian path (vanilla highway)
    - lane index wouldn't be an issue if we turn off vehicle/cims collisions (make them invisible for some time)

BTW I have noticed that scooter has 3 cims capacity? 😄
